src/QsCompiler/Transformations/FunctorGeneration.cs

```diff
@@ -148,6 +152,99 @@ public override QsStatementKind OnConjugation(QsConjugation stm)
         }
     }
 
+    /// <summary>
+    /// Scope transformation that splits any nested operation calls into separate statements so
+    /// that they can be properly reversed. This is necessary to avoid out of order execution of the
+    /// automatically generated adjoint. It is safe to do because an adjointable operation must return
+    /// Unit, so any nested calls can  be replaced by Unit and those calls moved to separate,
+    /// ordered statements.
+    /// </summary>
+    public class LiftOperationCalls
```

Comment:
nit: "Lift" is used elsewhere in the transformations to refer to when a piece of code is "lifted" into its own generated callable. Using that term here may cause confusion. Is there another term that could be used here? Maybe "Fan Out" or "Unnest"?

Reply:
That’s a good point, I don’t want to conflate the meaning. There’s got to be a term for this, but so far “unnest” is my favorite.

Reply:
At least one resource I found refers to this as “extracting” so maybe something like “ExtractNestedOperationCalls” would be nice and descriptive. 

https://refactoring.guru/extract-variable